### ISSUE TITLE
Ability to collect/count malformed rows in the DROPMALFORMED mode

### ISSUE THREAD

**Issue by ankurpathela** [databricks/spark-csv#327]
Need a way to be able to collect malformed rows in the csv. The DROPMALFORMED mode works great but gobbles the malformed rows, leaving the client without a way to report them or generate stats like error count.

A possible approach would be to introduce another mode, say REPORTMALFORMED, wherein the DataFrame is appended with a status column that contains one of the values amongst {PARSED, DROPPED}. For the rows which were dropped, the column values could be null/don't care.

**Comment by briggsw** [comment]
This seems very worthwhile. I'm not the most qualified person, but I'd be happy to take a stab at a pull request for this issue.

**Comment by HyukjinKwon** [comment]
Actually, for Spark-internal JSON source (Spark 2.0) will support `_corrupt_record` in `PERMISSIVE` mode. I think this feature might be helpful if added in CSV as well.

**Comment by falaki** [comment]
Agree. This would be great feature to have. Since spark-csv is nearing maintenance phase, I suggest we address this in upstream Apache Spark.

**Comment by ReymondRey** [comment]
Hello, 
I am faced to this exact problem of catching rejected lines after a dropmalformed. Does anyone have an answer or a tips on how ot do it ? That would be great :)

**Comment by BattleBeaver** [comment]
Hey guys, have you managed to find any workaround on this matter?

**Comment by alfredox10** [comment]
You may be able to create 2 dataframes and then subtract the resulting DROPMALFORMED from the original with all data such as...

df_bad = df_all.subtract(df_dropped_malformed)

**Comment by mathieu-rossignol** [comment]
Also needs a way to retrieve bad (skipped) lines. In production, this may help log the lines that where rejected from processing while injecting a new csv.

**Comment by mohantyashish109** [comment]
Has the feature been picked for any release. It would be really helpful to have this feature in.

**Comment by Mindful** [comment]
Would also really like to see this.

**Comment by PrashantShimpi** [comment]
@alfredox10 
I like the idea of trying to get subtract from df_all. 
but how u will be able to load the df_all as when reading the data you will be facing same issue
as you are trying to read same record.
